tuplex/codegen/src/LLVMEnvironment.cc

```diff
@@ -336,8 +336,12 @@ namespace tuplex {
                     memberTypes.push_back(doublePointerType());
                 } else if(elementType == python::Type::BOOLEAN) {
                     memberTypes.push_back(getBooleanPointerType());
-                } else if(elementType.isTupleType()) {
+                } else if(elementType.isTupleType() && elementType.isFixedSizeType()) {
```

Comment:
I would nest the if statements here: i.e.
```
if(elementType.isTupleType()) {
if elementType.isFixedSizeType()) {
...
} else {
...
}
}```

Reply:
Refactored.